app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -67,15 +67,31 @@ def reload_answer
   def publish_all
     graded_submissions = @assessment.submissions.with_graded_state
     if !graded_submissions.empty?
-      job =
-        Course::Assessment::Submission::PublishingJob.perform_later(@assessment, current_user).job
+      job = Course::Assessment::Submission::PublishingJob.
+            perform_later(@assessment, current_user).job
       redirect_to(job_path(job))
     else
       redirect_to course_assessment_submissions_path(current_course, @assessment),
                   notice: t('.notice')
     end
   end
 
+  # download either all of or a subset of submissions for an assessment.
+  def download_all
+    authorize!(:manage, @assessment)
```

Comment:
If you want to authorize it here, then this controller endpoint cannot be reused if students want to download a single submission. 

The alternative is to have 1 endpoint handle all possible cases of submission downloads for 1 assessment, then authorize based on the limit params.